docs/developer-guide.md

```diff
@@ -0,0 +1,84 @@
+# Pulsar Manager Developer's Guide
+
+Pulsar Manager is a web-based GUI management and monitoring tool that helps administrators and users manage and monitor tenants, namespaces, topics, subscriptions, brokers, clusters, and so on, and supports dynamic configuration of multiple environments.
+
+![architecture](img/pulsar-manager-architecture.png)
+
+The figure above illustrates how does Pulsar Manager work and how it manages multiple Pulsar environments. The Pulsar Manager comprises two parts: frontend and backend.
+
+## Backend
+
+Development tools used in the backend are listed below:
+
+* Spring-boot 
+* Mybatis
+* PageHelper
+* Swagger2
+* Zuul
+
+The code structure of the backend is shown below.
+
+![backend-code](img/pulsar-manager-backend-code.png)
+
+The backend configuration file is [here](https://github.com/streamnative/pulsar-manager/blob/master/src/main/resources/application.properties).
+The database schema is defined [here](https://github.com/streamnative/pulsar-manager/blob/master/src/main/resources/META-INF/sql/mysql-schema.sql)
+Add new RESTful api
+Restful api defines the specification. If the RESTful api needs backend processing, it needs to be prefixed with pulsar-manager. If you want to directly forward the request to the broker, you can directly define it according to the specification of Pulsar RESTful api.
+
+Let's take a new tenant as an example to illustrate how to develop in pulsar manager.
+1. Define a RESTful api
+Restful api that requires backend processing
+List all tenants
+GET /pulsar-manager/admin/v2/tenants
+     (2) Restful api directly forwarded to a broker
+Add a new tenant
+PUT /admin/v2/tenants/{tenantName}
+
```

Comment:
The format is incorrect, please adjust it.

![image](https://user-images.githubusercontent.com/50226895/64432870-f2284f80-d0ef-11e9-87ed-57a943836d80.png)